### PR TITLE
Implement VOR update steps to DocMap API

### PR DESCRIPTION
**Pull request by HazalCiplak**
https://github.com/elifesciences/data-hub-issues/issues/933

### CODE REVIEW

File: data_hub_api/docmaps/v2/api_input_typing.py [hunk 1]
```diff
@@ -35,6 +35,15 @@
     }
 )
 
+ApiVorVersionsInput = TypedDict(
```
Comment (de-code) [inline data_hub_api/docmaps/v2/api_input_typing.py]: minor: would rename to singular `ApiVorVersionInput` as it represents a single version

File: data_hub_api/docmaps/v2/codecs/docmaps.py [hunk 2]
```diff
@@ -284,11 +308,13 @@ def iter_docmap_steps_for_query_result_item(query_result_item: ApiInput) -> Iter
                     query_result_item=query_result_item,
                     manuscript_version=manuscript_version
                 )
-        if manuscript_version['vor_publication_date'] and index == len(manuscript_versions) - 1:
-            yield get_docmaps_step_for_vor_published_status(
-                query_result_item=query_result_item,
-                manuscript_version=manuscript_version
-            )
+        if vor_versions and index == len(manuscript_versions) - 1:
```
Comment (de-code) [inline data_hub_api/docmaps/v2/codecs/docmaps.py:311]: very minor: I'd maybe find it easier to follow if it was outside the loop. The if statement could then be something like (correct me if I am wrong):

```python
if manuscript_versions and vor_versions:
```

File: data_hub_api/docmaps/v2/codecs/elife_manuscript.py [hunk 3]
```diff
@@ -228,8 +228,19 @@ def get_docmap_elife_manuscript_output_for_published_step(
 
 
 def get_docmap_elife_manuscript_output_content_for_vor(
-    query_result_item: ApiInput
+    query_result_item: ApiInput,
+    vor_version_number: int
 ) -> Sequence[DocmapContent]:
+    if vor_version_number > 1:
```
Comment (de-code) [inline data_hub_api/docmaps/v2/codecs/elife_manuscript.py]: Doesn't that mean that version 1 would point to the article without version number. And that would usually redirect to the latest version?

Comment (HazalCiplak) [reply data_hub_api/docmaps/v2/codecs/elife_manuscript.py]: Double checking with Fred: https://github.com/elifesciences/data-hub-api/pull/177/files#r1661245261

**Review by de-code — APPROVED** [review]

**Pull request merged**